config/base/derive/src/configurable.rs

```diff
@@ -0,0 +1,228 @@
+        .zip(lvalue.iter())
+        .zip(option.iter())
+        .map(|((((ty, ident), &as_str_attr), l_value), &is_option)| {
+            let is_string = if let Type::Path(TypePath { path, .. }) = ty {
+                path.is_ident("String")
+            } else {
+                false
+            };
+            let set_field = if is_string {
+                quote! { #l_value = var }
+            } else if as_str_attr {
+                quote! {
+                    #l_value = serde_json::from_value(var.into())
+                        .map_err(|e| iroha_config_base::derive::Error::field_error(stringify!(#ident), e))?
+                }
+            } else if is_option {
+                quote! {
+                    #l_value = Some(serde_json::from_value(var.into())
+                        .map_err(|e| iroha_config_base::derive::Error::field_error(stringify!(#ident), e))?)
+                }
+            }
+            else {
+                quote! {
+                    #l_value = serde_json::from_str(&var)
+                        .map_err(|e| iroha_config_base::derive::Error::field_error(stringify!(#ident), e))?
+                }
+            };
+            (set_field, l_value, ty)
+        })
+        .zip(field_environment.iter())
+        .zip(inner.iter())
+        .zip(option.iter())
+        .map(|((((set_field, l_value, ty), field_env), &inner_thing), &is_option)| {
+            // let inner_thing2 = quote! {};
+            let inner_thing2 = if inner_thing && is_option {
+                let inner_ty = get_inner_type("Option", ty);
+                quote! {
+                    // #l_value = <#inner as iroha_config_base::LoadFromEnv>::load_environment().ok();
+                    // let inner_config = #inner::new();
+                    <#inner_ty as iroha_config_base::LoadFromEnv>::load_environment()?;
+                }
+            } else if inner_thing {
+                quote! {
+                    #l_value.load_environment()?;
+                }
+            }
+            else {
+                quote! {}
+            };
+            quote! {
+                if let Ok(var) = std::env::var(#field_env) {
+                    #set_field;
+                }
+                #inner_thing2
+            }
+        });
+
+    quote! {
+        impl iroha_config_base::proxy::LoadFromEnv for #struct_name {
+            type Error = iroha_config_base::derive::Error;
+            fn load_environment(
+                &'_ mut self
+            ) -> core::result::Result<(), iroha_config_base::derive::Error> {
+                #(#set_field)*
+                Ok(())
+            }
+        }
+    }
+}
+
+fn impl_build(
+    parent_name: &Ident,
+    proxy_name: &Ident,
+    field_idents: &[&Ident],
+    field_attrs: &[&[Attribute]],
+) -> proc_macro2::TokenStream {
+    let checked_fields = gen_none_fields_check(field_idents, field_attrs);
+
+    quote! {
+        impl iroha_config_base::proxy::Builder for #proxy_name {
```

Comment:
advise extracting the trait name as a constant. It might be moved to a different location, and updating a single constant is easier than find and replace.